ogre/src/OgreLidarVisual.cc

```diff
@@ -349,6 +349,11 @@ void OgreLidarVisual::Update()
       }
     }
 
+    // The newly created dynamic lines are having default visibility as true.
+    // The visibility needs to be set as per the current value after the new
+    // renderables are created.
+    this->SetVisible(this->Visible());
```

Comment:
This is a little bit weird. Are you setting the variable with the letter? I think it's because this line, right?
```
      T::SetVisible(_visible);
```

Reply:
I'm sorry I don't understand the question. 

I did the above as the visibility changes once the newly created  `DynamicRenderables`  are added to the LidarVisual, the default value of the visibility of the newly created renderable is `true`. Hence I am setting the current value of the visibility again to make sure that the newly added DyanmicRenderables do not have a separate value from the intended current value. On deleting the old ones and creating the new ones (which happens either during changing of visual type or changing the sensor topic or setting DisplayNonHitting false) it used to be visible even though the checkbox was off.